src/core/load-fragment.c

```diff
@@ -3194,6 +3194,67 @@ int config_parse_tasks_max(
         return 0;
 }
 
+int config_parse_delegate(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        CGroupContext *c = data;
+        int r;
+
+        /* We either accept a boolean value, which may be used to turn on delegation for all controllers, or turn it
+         * off for all. Or it takes a list of controller names, in which case we add the specified controllers to the
+         * mask to delegate. */
+
+        r = parse_boolean(rvalue);
+        if (r < 0) {
+                const char *p = rvalue;
+                CGroupMask mask = 0;
+
+                for (;;) {
+                        _cleanup_free_ char *word = NULL;
+                        CGroupController cc;
+
+                        r = extract_first_word(&p, &word, NULL, EXTRACT_QUOTES);
+                        if (r == 0)
+                                break;
+                        if (r == -ENOMEM)
+                                return log_oom();
+                        if (r < 0) {
+                                log_syntax(unit, LOG_ERR, filename, line, r, "Invalid syntax, ignoring: %s", rvalue);
```

Comment:
I was wondering if it'd better to make this fatal. If there's an error here, we fall back to the default, i.e. Delegate=no (unless overriding previous Delegate=yes, but let's ignore this case), so it's "safe". On the other hand, it'll be very confusing if one expects some controllers to be delegated and they are not, a clean failure might be better.

Reply:
I am not too concerned about this tbh, as it indicates a user unit file typo really, and is easily fixed

We can discuss making parse errors like this fatal, but that probably should happen in a separate PR, and be done for all extract_first_word() loops we have in load-fragment.c, not just this one.